### ISSUE TITLE
Evd throws an exception for diagonal matrices

### ISSUE THREAD

**Issue by mjmckp** [mathnet/mathnet-numerics#446]
The following F# code:
```
(DiagonalMatrix.ofDiagArray [|1.; 2.|]).Evd()
```
throws the exception:
```
System.IndexOutOfRangeException: Cannot set an off-diagonal element in a diagonal matrix.
   at MathNet.Numerics.LinearAlgebra.Storage.DiagonalMatrixStorage`1.At(Int32 row, Int32 column, T value)
   at MathNet.Numerics.LinearAlgebra.Double.Factorization.UserEvd.SymmetricTridiagonalize(Matrix`1 eigenVectors, Double[] d, Double[] e, Int32 order)
   at MathNet.Numerics.LinearAlgebra.Double.Factorization.UserEvd.Create(Matrix`1 matrix, Symmetricity symmetricity)
```
Instead, the `Evd()` routine should detect that the underlying matrix is diagonal, and return the trivial EVD.

**Comment by cdrnet** [comment]
Yes, we should do that, for all decompositions.

**Comment by cdrnet** [comment]
The underlying issue has been fixed in 8accea1. I'm thus closing this issue, but opened #472 to keep track that we should return the trivial decomposition directly. Thanks again for reporting!
